internal/host/validator.go

```diff
@@ -346,19 +346,16 @@ func (v *validator) isMtuValidAllInterfaces(c *validationContext, connectivityRe
 
 // isMtuValidInMachineNetwork - intended for CMN, focusing exclusively on the MTU reports for the machine network.
 func (v *validator) isMtuValidInMachineNetwork(c *validationContext, connectivityReport *models.ConnectivityReport) (ValidationStatus, string) {
-	for _, machineNet := range c.cluster.MachineNetworks {
-		_, mNetwork, err := net.ParseCIDR(string(machineNet.Cidr))
-		if err != nil {
-			return ValidationError, "Internal error - failed to parse machine network CIDR"
-		}
-		for _, r := range connectivityReport.RemoteHosts {
-			for _, mtuReport := range r.MtuReport {
-				sourceIP, err := network.FindSourceIPInMachineNetwork(mtuReport.OutgoingNic, mNetwork, c.inventory.Interfaces)
+	for _, r := range connectivityReport.RemoteHosts {
+		for _, mtuReport := range r.MtuReport {
+			remoteIP := net.ParseIP(mtuReport.RemoteIPAddress)
+			for _, machineNet := range c.cluster.MachineNetworks {
+				_, mNetwork, err := net.ParseCIDR(string(machineNet.Cidr))
 				if err != nil {
-					return ValidationError, err.Error()
+					return ValidationError, "Internal error - failed to parse machine network CIDR"
 				}
-				remoteIP := net.ParseIP(mtuReport.RemoteIPAddress)
-				if mNetwork.Contains(remoteIP) && sourceIP != "" {
+				// We are only interested in the remote IP within machine networks. If it is not part of any machine network, we don't need to consider it.
+				if mNetwork.Contains(remoteIP) {
```

Comment:
I think we should also verify the source IP belongs to any of the machine networks.